### PR TITLE
WFCORE-290 periodic and size rotating file handlers for management audit logging

### PR DESCRIPTION
**Pull request by istudens**
This introduces new rotating file handlers (size and periodic rotating handlers) for the management audit logging.

@kabir Could you review it please?

https://issues.jboss.org/browse/WFCORE-290

### CODE REVIEW

**Comment by jamezp** [comment]
I did not fully review this, but if we're going to continue to add handlers we need to look at how we could potentially tie audit logging handlers to the log manager handlers. Having to maintain the code in two places is not ideal.

File: domain-management/src/test/java/org/jboss/as/domain/management/security/auditlog/AbstractAuditLogHandlerTestCase.java [hunk 1]
```diff
@@ -216,12 +217,53 @@ protected ModelNode createAddFileHandlerOperation(String handlerName, String for
         return op;
     }
 
+    protected ModelNode createAddPeriodicRotatingFileHandlerOperation(String handlerName, String formatterName, String fileName, String suffix) {
+        ModelNode op = Util.createAddOperation(createPeriodicRotatingFileHandlerAddress(handlerName));
+        op.get(ModelDescriptionConstants.RELATIVE_TO).set("log.dir");
+        op.get(ModelDescriptionConstants.PATH).set(fileName);
+        op.get(ModelDescriptionConstants.FORMATTER).set(formatterName);
+        if (suffix != null)
```
Comment (kabir) [inline domain-management/src/test/java/org/jboss/as/domain/management/security/auditlog/AbstractAuditLogHandlerTestCase.java]: Nitpick, use braces

File: domain-management/src/test/java/org/jboss/as/domain/management/security/auditlog/AbstractAuditLogHandlerTestCase.java [hunk 2]
```diff
@@ -216,12 +217,53 @@ protected ModelNode createAddFileHandlerOperation(String handlerName, String for
...
+            op.get(ModelDescriptionConstants.SUFFIX).set(suffix);
+        return op;
+    }
+
+    protected ModelNode createAddSizeRotatingFileHandlerOperation(String handlerName, String formatterName, String fileName, String rotateSize, int maxBackupIndex) {
+        ModelNode op = Util.createAddOperation(createSizeRotatingFileHandlerAddress(handlerName));
+        op.get(ModelDescriptionConstants.RELATIVE_TO).set("log.dir");
+        op.get(ModelDescriptionConstants.PATH).set(fileName);
+        op.get(ModelDescriptionConstants.FORMATTER).set(formatterName);
+        if (rotateSize != null)
```
Comment (kabir) [inline domain-management/src/test/java/org/jboss/as/domain/management/security/auditlog/AbstractAuditLogHandlerTestCase.java]: Nitpick braces

**Comment by kabir** [comment]
This needs a test in org.jboss.as.core.model.test.auditlog in core-model-test to test the parsing.

**Comment by istudens** [comment]
@kabir parsing tests have been added and also those braces have been fixed ;)

**Pull request merged**